src/lib/registration/ui/migration_repos_workflow.rb

```diff
@@ -178,6 +175,23 @@ def registration_check
         register_system
       end
 
+      # check the current registration status
+      # @return [nil, Symbol] the workflow symbol (:next, :skip) or nil if not
+      #   in an installation
+      def registration_check_at_installation
+        # handle system upgrade (fate#323163)
+        return nil unless Yast::Stage.initial
+        # test autoupgrade first, Mode.update covers the autoupgrade as well
+        return registration_check_at_autoupgrade if Yast::Mode.autoupgrade
+        return system_upgrade_check if Yast::Mode.update
+
+        nil
```

Comment:
np: I think `nil` is not needed here. But certainly it makes the code easier to read.

Reply:
Yes, I wanted to make it clear that `nil` is returned at this point.